app/models/role.rb

```diff
@@ -17,6 +18,14 @@ def self.add_admin(person)
     find_or_create_by(person_id: person.id, name: "admin")
   end
 
+  def self.moderator?(person)
+    exists?(person_id: person.id, name: "moderator")
+  end
```

Comment:
Maybe we should just consider all admins moderators? So the scope would become `where(name: %w(moderator admin))`, this would become `moderators.exists?(person_id: person.id)`, the filter would become `redirect_unless_moderator` and reports would be mailed to `Role.moderators.map`